src/wasm-interpreter.h

```diff
@@ -2794,6 +2799,22 @@ template<typename GlobalManager, typename SubType> class ModuleInstanceBase {
                                        index.getSingleValue().geti32());
     }
 
+    Flow visitTableSet(TableSet* curr) {
+      NOTE_ENTER("TableSet");
+      Flow index = this->visit(curr->index);
+      if (index.breaking()) {
+        return index;
+      }
+      Flow flow = this->visit(curr->value);
+      if (flow.breaking()) {
+        return flow;
+      }
+      auto info = instance.getTableInterfaceInfo(curr->table);
+      info.interface->tableStore(
+        info.name, index.getSingleValue().geti32(), flow.getSingleValue());
+      return Flow();
+    }
```

Comment:
@kripken main spec tests are passes but validator got some errors in `ab.wast`:
```wat
[wasm-validator error in function 3] externref != i32: table.set index must be an i32, on 
(table.set $t2
 (local.get $0)
 (local.get $1)
)
[wasm-validator error in function 3] table.set value must have right type, on 
(table.set $t2
 (local.get $0)
 (local.get $1)
)
[wasm-validator error in function 4] funcref != i32: table.set index must be an i32, on 
(table.set $t3
 (local.get $0)
 (local.get $1)
)
[wasm-validator error in function 4] table.set value must have right type, on 
(table.set $t3
 (local.get $0)
 (local.get $1)
)
[wasm-validator error in function 5] funcref != i32: table.set index must be an i32, on 
(table.set $t3
 (local.get $0)
 (table.get $t3
  (local.get $1)
 )
)
[wasm-validator error in function 5] table.set value must have right type, on 
(table.set $t3
 (local.get $0)
 (table.get $t3
  (local.get $1)
 )
)
Fatal: error validating input
```

Any guesses as to why this is so?

Reply:
I'm not sure, but I would start with the first of those,
```
[wasm-validator error in function 3] externref != i32: table.set index must be an i32, on 
(table.set $t2
 (local.get $0)
 (local.get $1)
)
```
Is `$0` really an externref and not an i32?

When debugging this `BINARYEN_PRINT_FULL=1` in the env can help as it will annotate types next to things.

Reply:
Hmm, when I try to run `BINARYEN_PRINT_FULL=1 ./check.py spec` script very fast fail with this:
```bash
.. address-offset-range.fail.wast
executing:  <path>/binaryen/bin/wasm-shell <path>/binaryen/test/spec/address-offset-range.fail.wast
<< test failed as expected >>
.. address64.wast
executing:  <path>/binaryen/bin/wasm-shell <path>/binaryen/test/spec/address64.wast
executing:  <path>/binaryen/bin/wasm-shell <path>/binaryen/test/spec/address64.wast
        testing split module 0
executing:  /Volumes/Archive/Projects/Github/binaryen/bin/wasm-opt split.wast -O -all
warning: no output file specified, not emitting output
         (binary format check)
             <path>/binaryen/bin/wasm-as split.wast -o a.wasm -all -g
             <path>/binaryen/bin/wasm-dis a.wasm -o ab.wast -all
             <path>/binaryen/bin/wasm-opt ab.wast -all
[parse exception: expected list (at 37:2)]
Fatal: error parsing wasm
```